src/graphics/windowcontroller.cpp

```diff
@@ -1,7 +1,13 @@
 #include "windowcontroller.hpp"
 #include "../config/graphics.hpp"
 
-WindowController::WindowController() {}
+WindowController::WindowController() { }
+
+WindowController::~WindowController()
+{
+    glfwDestroyWindow(window);
+    glfwTerminate();
+}
 
 WindowController& WindowController::GetInstance()
```

Comment:
If we use a non-empty destructor, I don't think it's a good idea to make this static (possible problems with destruction order - I doubt it will happen in our case but if it doeas, it may be difficult to debug). Instead you could just explicitly create the object in main. In that case you could use its contructor to do all the glfwInit etc.